src/engraving/dom/textbase.cpp

```diff
@@ -1000,8 +1001,10 @@ void TextBlock::layout(const TextBase* t)
         }
         break;
         case ElementType::MEASURE: {
+            // ignore courtesy keysig, timesig, but fall back if needed
             Measure* m = toMeasure(e);
-            layoutWidth = m->ldata()->bbox().width();
+            const BarLine* bl = m->endBarLine();
+            layoutWidth = bl ? bl->segment()->x() + bl->ldata()->bbox().width() : m->width();
```

Comment:
and I'd rather not remove that `ldata()->bbox().` withoiut being 100% sure wherther and when it makes a difference

Reply:
Well, I meant rather for the else branch, to not break anything there

Reply:
Maybe also check a quite similar code in playbaccursor.cpp:
```cpp
        if (ns) {
            t2 = ns->tick();
            x2 = ns->canvasPos().x();
        } else {
            t2 = measure->endTick();
            // measure->width is not good enough because of courtesy keysig, timesig
            mu::engraving::Segment* seg = measure->findSegment(mu::engraving::SegmentType::EndBarLine, measure->tick() + measure->ticks());
            if (seg) {
                x2 = seg->canvasPos().x();
            } else {
                x2 = measure->canvasPos().x() + measure->width(); // safety, should not happen
            }
        }
```

Reply:
well yes, but that gives the left of the barline. and it uses `measure->width()`